### PR TITLE
Upgrade httpcomponents and consolidate

### PR DESCRIPTION
**Pull request by cbridgha**
Upgrading httpcomponents libraries to latest (4.4.13)
 - Also consolidating most references to the OSS lib, and switching all to use the wrapped bundle instead.
 - Moving up caused test dependency ripple affects including upgrading htmlunit framework, which in turn, brought in a few more new dependencies.
The htmlunit upgrade also required fixing deprecated api's in a few tests.

### CODE REVIEW

**Comment by cbridgha** [comment]
#build

**Comment by cbridgha** [comment]
#build

**Comment by cbridgha** [comment]
Asking for review from several SME's because this consolidation ended up touching so many projects.

**Comment by cbridgha** [comment]
Latest build has a good chance of resolving all test issues:  https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_L2kpwBhVEeuFUN9G2T7ZeA

**Review by jvanhill — APPROVED** [review]

**Review by pnicolucci — APPROVED** [review]
The jsf changes look reasonable here.

File: dev/com.ibm.ws.jsfContainer_fat_2.2/bnd.bnd [hunk 1]
```diff
@@ -33,5 +33,10 @@ fat.project: true
 	com.ibm.websphere.javaee.jsf.2.2;version=latest,\
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsfContainer_fat_2.2/bnd.bnd:33]: Should we update the copyright date range?

File: dev/com.ibm.ws.jsfContainer_fat_2.2/build.gradle [hunk 2]
```diff
@@ -15,12 +15,18 @@ configurations {
 }
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsfContainer_fat_2.2/build.gradle:15]: Should we update the copyright date range here?

File: dev/com.ibm.ws.jsfContainer_fat_2.3/bnd.bnd [hunk 3]
```diff
@@ -39,6 +39,11 @@ fat.project: true
 	com.ibm.websphere.javaee.jsf.2.3;version=latest,\
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsfContainer_fat_2.3/bnd.bnd:39]: same comment about the copyright

File: dev/com.ibm.ws.jsfContainer_fat_2.3/build.gradle [hunk 4]
```diff
@@ -16,15 +16,20 @@ configurations {
 }
 
 dependencies {
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsfContainer_fat_2.3/build.gradle:18]: same copyright range comment

**Review by andymc12 — APPROVED** [review]
LGTM - thanks @cbridgha!

File: dev/com.ibm.ws.jsfContainer_fat_2.3/build.gradle (hunk 4 above)
Comment (cbridgha) [reply dev/com.ibm.ws.jsfContainer_fat_2.3/build.gradle:18]: updated years

File: dev/com.ibm.ws.jsfContainer_fat_2.3/bnd.bnd (hunk 3 above)
Comment (cbridgha) [reply dev/com.ibm.ws.jsfContainer_fat_2.3/bnd.bnd:39]: updated years

File: dev/com.ibm.ws.jsfContainer_fat_2.2/bnd.bnd (hunk 1 above)
Comment (cbridgha) [reply dev/com.ibm.ws.jsfContainer_fat_2.2/bnd.bnd:33]: updated years

**Comment by cbridgha** [comment]
#libby

**Comment by cbridgha** [comment]
Next build: https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_ACIephiSEeuFUN9G2T7ZeA

**Comment by cbridgha** [comment]
#build

**Comment by cbridgha** [comment]
#build

**Comment by cbridgha** [comment]
#libby

**Pull request merged**